### ISSUE TITLE
Setting a nonexistent --kubernetes-version is overly sticky

### ISSUE THREAD

**Issue by 0bprashanthc** [kubernetes/minikube#3876]
Operating System: Mac OS (Mojave) 10.14.3 (18D109)
Steps Perfomed:
1. minikube start --vm-driver=xhyve
    Observation: Everything worked! I can see the cluster up and running with v1.13.4 K8s
2. minikube stop
    Observation: Stopped the cluster
3. minikube delete
   Observation: Deleted the cluster
4. Tried starting minikube again this time with version as "1.14.0". (I know it does not exist)
5. minikube start --vm-driver=xhyve --kubernetes-version="v1.14.0" 
Observation:
```
o   minikube v0.35.0 on darwin (amd64)
i   Tip: Use 'minikube start -p <name>' to create a new cluster, or 'minikube delete' to delete this one.
:   Re-using the currently running xhyve VM for "minikube" ...
:   Waiting for SSH access ...
-   "minikube" IP address is 192.168.64.3
-   Configuring Docker as the container runtime ...
-   Preparing Kubernetes environment ...
@   Downloading kubeadm v1.14.0
@   Downloading kubelet v1.14.0
!   Failed to update cluster: downloading binaries: downloading kubeadm: Error downloading kubeadm v1.14.0: failed to download: failed to download to temp file: download failed: 1 error(s) occurred:

* received invalid status code: 404 (expected 200)
```
6. This is expected. So this tried to start minikube with right kubernetes version "v1.13.4"
7. minikube start --vm-driver=xhyve --kubernetes-version="v1.13.4"
Observation:
```
o   minikube v0.35.0 on darwin (amd64)
i   Tip: Use 'minikube start -p <name>' to create a new cluster, or 'minikube delete' to delete this one.
:   Re-using the currently running xhyve VM for "minikube" ...
:   Waiting for SSH access ...
-   "minikube" IP address is 192.168.64.3
-   Configuring Docker as the container runtime ...
-   Preparing Kubernetes environment ...
@   Downloading kubeadm v1.14.0
@   Downloading kubelet v1.14.0
!   Failed to update cluster: downloading binaries: downloading kubeadm: Error downloading kubeadm v1.14.0: failed to download: failed to download to temp file: download failed: 1 error(s) occurred:

* received invalid status code: 404 (expected 200)
```

8. Not sure why it is picking 1.14.0 when it is different than what I asked for
9. I tried starting minikube without version (basically step 1 again)
10. minikube start --vm-driver=xhyve
Observation:
```
o   minikube v0.35.0 on darwin (amd64)
i   Tip: Use 'minikube start -p <name>' to create a new cluster, or 'minikube delete' to delete this one.
:   Re-using the currently running xhyve VM for "minikube" ...
:   Waiting for SSH access ...
-   "minikube" IP address is 192.168.64.3
-   Configuring Docker as the container runtime ...
-   Preparing Kubernetes environment ...
@   Downloading kubeadm v1.14.0
@   Downloading kubelet v1.14.0
!   Failed to update cluster: downloading binaries: downloading kubeadm: Error downloading kubeadm v1.14.0: failed to download: failed to download to temp file: download failed: 1 error(s) occurred:

* received invalid status code: 404 (expected 200)
```
Basically my minikube crashed! Not sure why is this though!

Any help would be appreciated.

**Comment by tstromberg** [comment]
This sounds like two bugs:

- Once you start a cluster to a newer Kubernetes version, minikube won't downgrade it. It's supposed to output a warning message, but I don't see it here. 
- If you configure a cluster to a non-working Kubernetes version, we save it to the configuration and persist that configuration across restarts.

As the message says, to get around this you can start a new cluster, or delete this one:

`i Tip: Use 'minikube start -p ' to create a new cluster, or 'minikube delete' to delete this one.`

**Comment by afbjorklund** [comment]
I wrote some code to validate against available Kubernetes versions. I should revive that...

**Comment by tstromberg** [comment]
I believe this was fixed in minikube v1.4.
